frontend/src/util/cardHelpers.ts

```diff
@@ -0,0 +1,66 @@
+// prettier-ignore
+type Rank = (
+  | 'A' | '2' | '3' | '4' | '5' | '6' | '7'
+  | '8' | '9' | 'T' | 'J' | 'Q' | 'K'
+);
+type Suit = 'diamonds' | 'clubs' | 'hearts' | 'spades';
+
+export type SuitCard = {
+  type: 'suit_card';
+  rank: Rank;
+  suit: Suit;
+};
+
+type Card = SuitCard | {type: 'big_joker'} | {type: 'little_joker'};
+
+// prettier-ignore
+const orderedRanks: (Rank | null)[] = [
+  'A', '2', '3', '4', '5', '6', '7',
+  '8', '9', 'T', 'J', null, 'Q', 'K',
```

Comment:
I see that you found the "knight" thing in unicode. lol.

Rather than implementing this stuff here, though, can we add additional information as necessary to cards.js, and pass the information from the backend? Having two unicode<->card<->suit mappings feels not ideal.

Reply:
Removed unicode stuff in 638245f6d3464a95ddc717139e42cc100f71a532

Reply:
Is there a way to generate cards.js as a file that can be checked in? any code referencing this cannot run in unit tests

Reply:
yeah, we can do that. maybe add a Rust test which asserts the checked in version and the generated version match, too.